.github/workflows/test-terra-jupyter-aou.yml

```diff
@@ -0,0 +1,112 @@
+name: Test terra-jupyter-aou
+# Perform smoke tests on the terra-jupyter-aou Docker image to have some amount of confidence that
+# Python package versions are compatible.
+#
+# To configure the minimal auth needed for these tests to be able to read public data from Google Cloud Platform:
+# Step 1: Create a service account per these instructions:
+#         https://github.com/google-github-actions/setup-gcloud/blob/master/setup-gcloud/README.md
+# Step 2: Give the service account the following permissions within the project: BigQuery User
+# Step 3: Store its key and project id as GitHub repository secrets GCP_SA_KEY and GCP_PROJECT_ID.
+#         https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository
+
+on:
+  pull_request:
+    branches: [ master ]
+    paths:
+    - 'terra-jupyter-aou/**'
+    # Note: secrets are not passed to pull requests from forks, so the dev team will need to use the manual workflow
+    # dispatch trigger when receiving community contributions.
+
+  workflow_dispatch:
+    # Allows manually triggering of workflow on a selected branch via the GitHub Actions tab.
+    # GitHub blog demo: https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/.
+    
+  push:
+    # TODO(deflaux) remove these 'push' triggers after testing is complete.
+    branches: [ add-smoke-tests, add-aou-smoke-tests, bump-aou-versions ]
```

Comment:
can you remove these if you're ready for this to be merged

Reply:
I'll update the workflow to conditionally build the base image(s), if it doesn't exist. And then for the purpose of the test, there will be a local edit to the Dockerfile 'FROM' clause to refer to the local-only base image.

will do on removing the test branch triggers

Reply:
Okay, this now builds the base images recursively, when missing.

* [workflow result](https://github.com/deflaux/terra-docker/actions/runs/556319095) when base image for terra-jupyter-python is missing
* [workflow result](https://github.com/deflaux/terra-docker/actions/runs/556319094) when base image for terra-jupyter-aou AND base image for terra-jupyter-python are both missing 
* Here’s how I tested missing base images: [test branch diff](https://github.com/deflaux/terra-docker/commit/0b9d98332c15f0601e06a4ae67480f811e6a9b0b)
   * You’ll see in this diff that I also had to comment out JupyterLab extension setup in terra-jupyter-base.
   * That section appears to be [failing](https://github.com/deflaux/terra-docker/runs/1875710096?check_suite_focus=true) in the current base image.

If this looks good to you folks now, I'll go ahead and remove those test branch triggers.